subsys/testsuite/ztest/include/zephyr/ztest.h

```diff
@@ -28,6 +28,12 @@
 #error "You need to add CONFIG_ZTEST to your config file."
 #endif
 
+#if defined(CONFIG_NO_OPTIMIZATIONS) && !defined(CONFIG_ZTEST_ALLOW_NO_OPTIMIZATIONS)
+#error Running tests with CONFIG_NO_OPTIMIZATIONS is not supported and \
+	known to break many tests, please do not file issues about it. If you \
+	really mean to do this also set CONFIG_ZTEST_ALLOW_NO_OPTIMIZATIONS.
+#endif
+
```

Comment:
Why can't we error in cmake? It seems that this is statically known.

Reply:
Grand idea, moved to a CMake message.